packages/front-end/components/Layout/SidebarLink.tsx

```diff
@@ -75,6 +65,35 @@ const SidebarLink: FC<SidebarLinkProps> = (props) => {
     return null;
   }
 
+  const permittedSubLinks: SidebarLinkProps[] = [];
+
+  if (props.subLinks) {
+    props.subLinks.forEach((l) => {
+      if (l.superAdmin && !superAdmin) return null;
+
+      if (l.multiOrgOnly && !isMultiOrg()) {
+        return null;
+      }
+      if (l.cloudOnly && !isCloud()) {
+        return null;
+      }
+      if (l.selfHostedOnly && isCloud()) {
+        return null;
+      }
+      if (l.permissionCallbacks) {
+        if (l.permissionCallbacks.every((cb) => cb())) {
+          permittedSubLinks.push(l);
+        }
+      } else {
+        permittedSubLinks.push(l);
+      }
+    });
+  }
+
+  if (props.subLinks && !permittedSubLinks.length) {
+    return null;
+  }
```

Comment:
This is the logic referenced above - if the parent link contains sublinks, but the user doesn't have permission for any sublinks, we don't render anything (parent or sublinks)